docs/source/developers/python.rst

```diff
@@ -138,10 +138,18 @@ You should now see
 Using Conda
 ~~~~~~~~~~~
 
+.. note::
+
+   Using conda to build Arrow on macOS is complicated by the
+   fact that the `conda-forge compilers require an older macOS SDK <https://stackoverflow.com/a/55798942>`_.
+   Conda offers some `installation instructions <https://docs.conda.io/projects/conda-build/en/latest/resources/compiler-tools.html#macos-sdk>`_;
+   the alternative would be to use :ref:`Homebrew <python-homebrew>` and
+   ``pip`` instead.
+
 Let's create a conda environment with all the C++ build and Python dependencies
 from conda-forge, targeting development for Python 3.7:
 
-On Linux and OSX:
+On Linux and macOS:
```

Comment:
I'd separate it for macos, with different snippets. One with compilers and mentioning it requires the old SDK and one without the compilers package.

Reply:
Given Uwe's last comment, it doesn't sound like we can do without compilers on macOS without some additional work. I'll make a jira for that and we can circle back later.

Reply:
Here's the followup jira: https://issues.apache.org/jira/browse/ARROW-5280